install/infra/single-cluster/k3s/variables.tf

```diff
@@ -0,0 +1,48 @@
+variable "kubeconfig" {
+  description = "The KUBECONFIG file path to store the resulting KUBECONFIG file to"
+  default     = "./kubeconfig"
+}
+
+variable "project" {
+  description = "Google cloud Region to perform operations in"
+}
+
+variable "region" {
+  description = "Google cloud Region to perform operations in"
+  default     = "europe-west1"
+}
+
+variable "zone" {
+  description = "Google cloud Zone to perform operations in"
+  default     = "europe-west1-b"
+}
+
+variable "credentials" {
```

Comment:
In a prior pull request we started passing credentials as a variable; this reduces the dependency on having files on-disk but this variable assumes that we're passing in the credentials directly.

Inserting the credentials into `terraform.tfvars` doesn't work well as the JSON strings need to have their double quotes escaped and terraform doesn't like the newlines in the JSON document.

WDYT of having this as `credentials_path` and then using something like the following:?

```
locals {
  credentials = "${file(var.credentials_path)}"
}
```

Is this a better approach or do you believe it'll be cheaper to switch back to passing around credentials by a file path?

Reply:
Great point!! has updated accordingly!